test/integ/test_cursor.py

```diff
@@ -1318,6 +1318,64 @@ def test_resultbatch_lazy_fetching(conn_cnx, result_format, patch_path):
                 assert patched_download.call_count == 5
 
 
+@pytest.mark.skipolddriver(reason="new feature in v2.5.0")
+@pytest.mark.parametrize(
+    "result_format,patch_path",
+    (
+        ("json", "snowflake.connector.result_batch.JSONResultBatch.create_iter"),
+        ("arrow", "snowflake.connector.result_batch.ArrowResultBatch.create_iter"),
+    ),
+)
+def test_resultbatches_can_access_schema(conn_cnx, result_format, patch_path):
+    rowcount = 1000000
+    with conn_cnx(
+        session_parameters={"python_connector_query_result_format": result_format}
+    ) as con:
+        with con.cursor() as cur:
+            cur.execute(
+                f"select seq4() as c1, randstr(1,random()) as c2 "
+                f"from table(generator(rowcount => {rowcount}));"
+            )
+            result_batches = cur.get_result_batches()
+            batch_schemas = [batch.schema for batch in result_batches]
+            for schema in batch_schemas:
+                # 2 columns in result
+                assert len(schema) == 2
+                # all batches should have the same schema
+                assert schema[0] == ResultMetadata("C1", 0, None, None, 10, 0, False)
+                assert schema[1] == ResultMetadata(
+                    "C2", 2, None, 16777216, None, None, False
+                )
+
+
+@pytest.mark.skipolddriver(reason="new feature in v2.5.0")
+@pytest.mark.parametrize(
+    "result_format,patch_path",
+    (
+        ("json", "snowflake.connector.result_batch.JSONResultBatch.create_iter"),
+        ("arrow", "snowflake.connector.result_batch.ArrowResultBatch.create_iter"),
+    ),
```

Comment:
`patch_path` is not used
```suggestion
    "result_format", ("json","arrow"),
```